tests/Doctrine/Tests/DBAL/Functional/Types/DoubleTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Types;
+
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function abs;
+use function floatval;
+use function is_int;
+use function is_string;
+use function microtime;
+use function sprintf;
+
+class DoubleTest extends DbalFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $table = new Table('double_table');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('val', 'float');
+        $table->setPrimaryKey(['id']);
+
+        $sm = $this->connection->getSchemaManager();
+        $sm->dropAndCreateTable($table);
+    }
+
+    public function testInsertAndSelect() : void
+    {
+        $value1 = 1.1;
+        $value2 = 77.99999999999;
+        $value3 = microtime(true);
+
+        $this->insert(1, $value1);
+        $this->insert(2, $value2);
+        $this->insert(3, $value3);
+
+        $result1 = $this->select(1);
+        $result2 = $this->select(2);
+        $result3 = $this->select(3);
+
+        if (is_string($result1)) {
+            $result1 = floatval($result1);
+            $result2 = floatval($result2);
+            $result3 = floatval($result3);
+        }
+
+        if ($result1 === false) {
+            $this->fail('Expected $result1 to not be false');
+
+            return;
+        }
+        if ($result2 === false) {
+            $this->fail('Expected $result2 to not be false');
+
+            return;
+        }
+        if ($result3 === false) {
+            $this->fail('Expected $result3 to not be false');
+
+            return;
+        }
+
+        $diff1 = abs($result1 - $value1);
+        $diff2 = abs($result2 - $value2);
+        $diff3 = abs($result3 - $value3);
+
+        $this->assertLessThanOrEqual(0.0001, $diff1, sprintf('%f, %f, %f', $diff1, $result1, $value1));
+        $this->assertLessThanOrEqual(0.0001, $diff2, sprintf('%f, %f, %f', $diff2, $result2, $value2));
+        $this->assertLessThanOrEqual(0.0001, $diff3, sprintf('%f, %f, %f', $diff3, $result3, $value3));
+
+        $result1 = $this->selectDouble($value1);
+        $result2 = $this->selectDouble($value2);
+        $result3 = $this->selectDouble($value3);
+
+        $this->assertSame(is_int($result1) ? 1 : '1', $result1);
```

Comment:
There shouldn't be any conditionals in a test.

Reply:
@morozov 

Similar to my other comment, some drivers seem to return integers as strings in which case I have to either cast everything, or just detect.

I'm open to alternatives?